zcash_client_backend/proto/service.proto

```diff
@@ -8,18 +8,31 @@ option go_package = "lightwalletd/walletrpc";
 option swift_prefix = "";
 import "compact_formats.proto";
 
+// An identifier for a Zcash value pool.
+enum PoolType {
+    TRANSPARENT = 0;
+    SAPLING = 1;
+    ORCHARD = 2;
+}
+
 // A BlockID message contains identifiers to select a block: a height or a
 // hash. Specification by hash is not implemented, but may be in the future.
 message BlockID {
-     uint64 height = 1;
-     bytes hash = 2;
+    uint64 height = 1;
+    bytes hash = 2;
```

Comment:
There's something deeply annoying here: because the `BlockID` type is used directly as the argument to a GRPC request, and `BlockRange` specifies two `BlockID`s, we can't add the set of pool types here without the meaning of `BlockRange` becoming confusing. 

GRPC interfaces should be defined in terms of explicit `Request` types, so that additive changes of this sort are not invasive. Of course, `BlockID` should have been defined as a `oneof` in the first place, but that's a separate complaint. Really we should overhaul this whole interface, and create a `V2` API endpoint service type that does things properly.